test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/PropertiesTest.scala

```diff
@@ -173,4 +175,285 @@ class PropertiesTest {
+    val out1 = storeStream(prop1)
+    assertTrue(out1.toString().trim().endsWith("""Property\ C=see"""))
+
+    val prop2 = new Properties()
+    prop2.put("Property B", "b ee#!=:")
+    val out2 = storeStream(prop2)
+    assertTrue(out2.toString().trim().endsWith("""Property\ B=b ee\#\!\=\:"""))
+
+    val prop3 = new Properties()
+    prop3.put("Property A", " aye\\\f\t\n\r\b")
+    val out3 = storeStream(prop3)
+    // JVM outputs \b as \u0008 and you can't just add \u0008 to the end
+    val result = new StringBuilder("""Property\ A=\ aye\\\f\t\n\r""")
+      .appendAll(Array('\\', 'u', '0', '0', '0', '8'))
+    assertTrue(out3.toString().trim().endsWith(result.toString))
+  }
+
+  @Test def storeWriterComments(): Unit = {
+    val prop1 = new Properties()
+    prop1.put("Property A", " aye\\\f\t\n\r\b")
+    prop1.put("Property B", "b ee#!=:")
+    prop1.put("Property C", "see")
+    val out = storeWriter(prop1, "A Header")
+    val prop2 = loadByteArrayOutputStream(out)
+    assertAll(prop1, prop2)
+  }
+
+  // helper functions
+
+  def storeStream(props: Properties,
+      header: String = ""): ByteArrayOutputStream = {
+    val out = new ByteArrayOutputStream()
+    props.store(out, header)
+    out
+  }
+
+  def storeWriter(props: Properties,
+      header: String = ""): ByteArrayOutputStream = {
+    val out = new ByteArrayOutputStream()
+    props.store(new OutputStreamWriter(out), header)
+    out.close()
+    out
+  }
+
+  def loadByteArrayOutputStream(out: ByteArrayOutputStream): Properties = {
+    val props = new Properties()
+    props.load(new ByteArrayInputStream(out.toByteArray))
+    props
+  }
+
+  def assertAll(expected: Properties, actual: Properties): Unit = {
+    val e = expected.propertyNames()
+    while (e.hasMoreElements) {
+      val nextKey = e.nextElement().asInstanceOf[String]
+      assertEquals(actual.getProperty(nextKey), expected.getProperty(nextKey))
+    }
+  }
+
+  def loadStream(in: String): Properties = {
+    val prop = new java.util.Properties()
+    prop.load(new ByteArrayInputStream(
+        in.getBytes(StandardCharsets.ISO_8859_1)))
+    prop
+  }
+
+  def loadReader(in: String): Properties = {
+    val prop = new java.util.Properties()
+    prop.load(new InputStreamReader(new ByteArrayInputStream(
+        in.getBytes(StandardCharsets.UTF_8))))
+    prop
+  }
+
+  // scalastyle:off
+  lazy val filestr =
+    """|
+       |   \ \r \n \t \f
+       |
+       |! dshfjklahfjkldashgjl;as
+       |     #jdfagdfjagkdjfghksdajfd
+       |
+       |!!properties
+       |
+       |a=a
+       |b bb as,dn   
+       |c\r\ \t\nu =:: cu
+       |bu= b\
+       |		u
+       |d=d\r\ne=e
+       |f   :f\
+       |f\
+       |			f
+       |g		g
+       |h\ h
+       |\   i=i
+       |j=\   j
+       |space=\   c
+       |
+       |dblbackslash=\\
+       |
+       |# jvm does not trim trailing space so no line continuation
+       |trailing = foo, \  
+       |bar
+       |notrailing = baz \\  
```

Comment:
Here is one way to get rid of the trailing whitespace while preserving them, and in a way that makes it more clearly visible:
```suggestion
       |# jvm does not trim trailing space so no line continuation
       |trailing = foo, \${"  "}
       |bar
       |notrailing = baz \\${"  "}
```

Reply:
I used the method and also for the embedded tabs above.